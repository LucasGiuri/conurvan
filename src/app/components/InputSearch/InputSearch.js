import React from 'react';
import PropTypes from 'prop-types';


const inputSearch = ({placeholder, autoFocus, onChange}) => {
  return (
      <input autoFocus={autoFocus} type="search" placeholder={placeholder} onChange={onChange} />
  )
};

inputSearch.propTypes = {
  autoFocus: PropTypes.bool,
  placeholder: PropTypes.string,
  onChange: PropTypes.func.isRequired
};

inputSearch.defaultProps = {
  autoFocus: false,
  placeholder: ''
};

export default inputSearch;
