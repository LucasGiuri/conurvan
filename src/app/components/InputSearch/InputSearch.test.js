import React from 'react';
import InputSearch from './InputSearch';
import renderer from 'react-test-renderer';
import CloseIcon from '@/assets/icons/close_black.svg?react';

describe('InputSearch', () => {
	it('Match to snapshot when it has no params', () => {
    const component = renderer.create(
      <InputSearch />,
		);

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
	});

	it('Match to snapshot when we pass a value param', () => {
		const value = '21 BUTTONS';

		const component = renderer.create(
			<InputSearch value={value} />,
		);

		let tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('Match to snapshot when we pass a placeholder param', () => {
		const placeholder = '21 BUTTONS';
		const component = renderer.create(
			<InputSearch placeholder={placeholder} />,
		);

		let tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('Match to snapshot when is focused', () => {
		const component = renderer.create(
			<InputSearch autoFocus={true} />,
		);

		let tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('Match to snapshot when close button is hidden', () => {
		const component = renderer.create(
			<InputSearch isCleanable={false} />,
		);

		let tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('Match to snapshot when close button is show', () => {
		const component = renderer.create(
			<InputSearch isCleanable={true} />,
		);

		let tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('Trigger on change when user writes something', () => {
		const value = "21 BUTTONS";
		const onChange = jest.fn();

		const component = renderer.create(
			<InputSearch value={value} onChange={onChange} />,
		);

		const input = component.root.findByType('input');

		input.props.onChange();

		expect(onChange).toHaveBeenCalledTimes(1);
		expect(onChange).toHaveBeenCalledWith();
	});

	it('Clean input when user clicks clean button', () => {
		const onClick = jest.fn();

		const component = renderer.create(
			<InputSearch isCleanable={true} onClose={onClick} />,
		);

		const div = component.root.findByType(CloseIcon);
		div.props.onClick();

		expect(onClick).toHaveBeenCalledTimes(1);
		expect(onClick).toHaveBeenCalledWith();
	});
});
