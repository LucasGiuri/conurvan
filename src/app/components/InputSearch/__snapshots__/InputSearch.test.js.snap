// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputSearch Match to snapshot when close button is hidden 1`] = `
<div
  className="input-search"
>
  <search-button.svg
    className="input-search__icon"
  />
  <input
    className="input-search__input"
    type="text"
  />
</div>
`;

exports[`InputSearch Match to snapshot when close button is show 1`] = `
<div
  className="input-search"
>
  <search-button.svg
    className="input-search__icon"
  />
  <input
    className="input-search__input"
    type="text"
  />
  <close_black.svg
    className="input-search__icon"
  />
</div>
`;

exports[`InputSearch Match to snapshot when is focused 1`] = `
<div
  className="input-search"
>
  <search-button.svg
    className="input-search__icon"
  />
  <input
    autoFocus={true}
    className="input-search__input"
    type="text"
  />
</div>
`;

exports[`InputSearch Match to snapshot when it has no params 1`] = `
<div
  className="input-search"
>
  <search-button.svg
    className="input-search__icon"
  />
  <input
    className="input-search__input"
    type="text"
  />
</div>
`;

exports[`InputSearch Match to snapshot when we pass a placeholder param 1`] = `
<div
  className="input-search"
>
  <search-button.svg
    className="input-search__icon"
  />
  <input
    className="input-search__input"
    placeholder="21 BUTTONS"
    type="text"
  />
</div>
`;

exports[`InputSearch Match to snapshot when we pass a value param 1`] = `
<div
  className="input-search"
>
  <search-button.svg
    className="input-search__icon"
  />
  <input
    className="input-search__input"
    type="text"
    value="21 BUTTONS"
  />
</div>
`;
